### PR TITLE
Add support for the placeholders with precision

### PR DESCRIPTION
**Pull request by amieiro**
Add support for the placeholders with a format like %1$.2f

See https://meta.trac.wordpress.org/ticket/5927

### CODE REVIEW

File: tests/phpunit/testcases/test_builtin_warnings.php [hunk 1]
```diff
@@ -201,7 +201,18 @@ function test_placeholders() {
...
+		$this->assertNoWarnings( 'placeholders', 'Data: %1$.2fMB | Index: %2$.2fMB | Free: %3$.2fMB | Engine: %4$s', 'Data: %1$.2fMB | Index: %2$.2fMB | Free: %3$.2fMB | Engine: %4$s' );
+
+		$this->assertHasWarningsAndContainsOutput( 'placeholders',
+			'%1$.2f baba',
+			'%1$.f баба',
+		'Missing %1$.2f placeholder in translation.');
+		$this->assertHasWarningsAndContainsOutput( 'placeholders',
+			'%1$.f baba',
+			'%1$.4f баба',
+		'Extra %1$.4f placeholder in translation.');
```
Comment (ocean90) [inline tests/phpunit/testcases/test_builtin_warnings.php]: Can you make sure these lines use the same code style as the following lines?

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: tests/phpunit/testcases/test_builtin_warnings.php (hunk 1 above)
Comment (amieiro) [reply tests/phpunit/testcases/test_builtin_warnings.php]: Updated.

Comment (ocean90) [reply tests/phpunit/testcases/test_builtin_warnings.php]: Still doesn't look right: https://github.com/GlotPress/GlotPress-WP/blob/8a91ee7d9fb9c39359d6d8d09ace3868efa4221a/tests/phpunit/testcases/test_builtin_warnings.php#L208-L224

Comment (amieiro) [reply tests/phpunit/testcases/test_builtin_warnings.php]: Sorry. As I use PhpStorm with parameter name hints enabled, I didn't see this difference. I will update soon. 

image

File: tests/phpunit/testcases/test_builtin_warnings.php [hunk 2]
```diff
@@ -201,7 +201,20 @@ function test_placeholders() {
...
 		$this->assertNoWarnings( 'placeholders', '%1$d baba %2$@', '%2$@ баба %1$d' );
+		$this->assertNoWarnings( 'placeholders', '%1$.4f baba', '%1$.4f баба' );
+		$this->assertNoWarnings( 'placeholders', '%2$.2fMB baba', '%2$.2fMB баба' );
+		$this->assertNoWarnings( 'placeholders', 'Data: %1$.2fMB | Index: %2$.2fMB | Free: %3$.2fMB | Engine: %4$s', 'Data: %1$.2fMB | Index: %2$.2fMB | Free: %3$.2fMB | Engine: %4$s' );
 
+		$this->assertHasWarningsAndContainsOutput(
+			'placeholders',
+			'%1$.2f baba',
+			'%1$.f баба',
+			'Missing %1$.2f placeholder in translation.');
```
Comment (ocean90) [inline tests/phpunit/testcases/test_builtin_warnings.php]:
```suggestion
			'Missing %1$.2f placeholder in translation.'
		);
```

File: tests/phpunit/testcases/test_builtin_warnings.php [hunk 3]
```diff
@@ -201,7 +201,20 @@ function test_placeholders() {
...
+		$this->assertHasWarningsAndContainsOutput(
+			'placeholders',
+			'%1$.f baba',
+			'%1$.4f баба',
+			'Extra %1$.4f placeholder in translation.');
```
Comment (ocean90) [inline tests/phpunit/testcases/test_builtin_warnings.php]:
```suggestion
			'Extra %1$.4f placeholder in translation.'
		);
```

**Review by ocean90 — CHANGES_REQUESTED** [review]

**Pull request merged**